### PR TITLE
Feat/x2earn rewards pool

### PR DESCRIPTION
**Pull request by Agilulfo1820**
- Distribute rewards through X2EarnRewardsPool contract
- Aligned contract logic to new distribution method
- Mock VeBetterDAO contracts on solo / hardhat

### CODE REVIEW

File: apps/contracts/contracts/EcoEarn.sol [hunk 1]
```diff
@@ -26,14 +26,24 @@
...
+import '@openzeppelin/contracts/utils/Strings.sol';
 
 /**
  * @title EcoEarn Contract
  * @dev This contract manages a reward system based on cycles. Participants can make valid submissions to earn rewards.
+ * Rewards are being distributed by interacting with the VeBetterDAO's X2EarnRewardsPool contract.
+ *
+ * @notice To distribute rewards this contract necesitates of a valid APP_ID provided by VeBetterDAO. This contract
+ * can be initially deployed without this information and DEFAULT_ADMIN_ROLE can update it later through {EcoEarn-setAppId}.
+ * This contract must me set as a `rewardDistributor` inside the X2EarnApps contract to be able to send rewards to users and withdraw.
```
Comment (pierobassa) [inline apps/contracts/contracts/EcoEarn.sol:39]: `be` typo

File: apps/contracts/contracts/EcoEarn.sol [hunk 2]
```diff
@@ -106,21 +121,20 @@ contract EcoEarn is AccessControl {
         // Decrease the rewards left
         rewardsLeft[getCurrentCycle()] -= amount;
 
-        // Transfer the reward to the participant
-        require(token.transfer(participant, amount));
+        // Transfer the reward to the participant, will revert if the transfer fails
+        x2EarnRewardsPoolContract.distributeReward(appId, amount, participant, '');
```
Comment (pierobassa) [inline apps/contracts/contracts/EcoEarn.sol:125]: Shall we also specify that last param is a string for the proof and optional? Pointing to info on how the proof should be formatted

**Review by pierobassa — COMMENTED** [review]
Great stuff! I'd also extend deployments to testnet too or add in the readme how to interact with existing testnet contracts through the dev environment

majority of devs go through testnet and not solo

File: apps/contracts/contracts/EcoEarn.sol (hunk 1 above)
Comment (Agilulfo1820) [reply apps/contracts/contracts/EcoEarn.sol:39]: Yes, in testenet devs should interact with dev environment. Will add to docs.

File: apps/contracts/contracts/EcoEarn.sol (hunk 2 above)
Comment (Agilulfo1820) [reply apps/contracts/contracts/EcoEarn.sol:125]: Sure!

File: apps/contracts/contracts/EcoEarn.sol (hunk 1 above)
Comment (Agilulfo1820) [reply apps/contracts/contracts/EcoEarn.sol]: Done

File: apps/contracts/contracts/EcoEarn.sol (hunk 2 above)
Comment (Agilulfo1820) [reply apps/contracts/contracts/EcoEarn.sol:130]: Done

**Review by pierobassa — APPROVED** [review]

**Pull request merged**